lib/samples/find_address_with_reverse_geocode/find_address_with_reverse_geocode_sample.dart

```diff
@@ -46,158 +49,84 @@ class _FindAddressWithReverseGeocodeSampleState
-    _geocodeParameters.resultAttributeNames
-        .addAll(['Place_addr', 'Match_addr']);
-
-    await Future.wait([
-      _worldLocatorTask.load(),
-      _sanDiegoLocatorTask.load(),
-    ]);
+    await _worldLocatorTask.load();
 
     setState(() => _ready = true);
   }
 
   void onTap(Offset localPosition) async {
-    final identifyGraphicsOverlayResult =
-        await _mapViewController.identifyGraphicsOverlay(
-      _graphicsOverlay,
-      screenPoint: localPosition,
-      tolerance: 5,
-    );
+    // remove already existing graphics.
+    if (_graphicsOverlay.graphics.isNotEmpty) _graphicsOverlay.graphics.clear();
 
-    if (identifyGraphicsOverlayResult.graphics.isEmpty) return;
+    // convert the screen point to a map point.
+    final mapTapPoint =
+        _mapViewController.screenToLocation(screen: localPosition);
+    if (mapTapPoint == null) return;
 
-    if (mounted) {
-      final graphic = identifyGraphicsOverlayResult.graphics.first;
-      final center = graphic.geometry!.extent.center;
-      final x = center.x.toStringAsFixed(3);
-      final y = center.y.toStringAsFixed(3);
-      final matchAddr = graphic.attributes['Match_addr'];
-      showDialog(
-        context: context,
-        builder: (BuildContext context) {
-          return AlertDialog(content: Text('$matchAddr\n($x, $y)'));
-        },
-      );
-    }
-  }
+    // normalize point.
+    final normalizedTapPoint =
+        GeometryEngine.normalizeCentralMeridian(geometry: mapTapPoint);
+    if (normalizedTapPoint == null) return;
 
-  void onServiceChanged(LocatorTask? locatorTask) {
-    if (locatorTask == null) return;
+    // create a graphic object for the specified point.
+    _graphicsOverlay.graphics.add(Graphic(geometry: normalizedTapPoint));
 
-    setState(() => _locatorTask = locatorTask);
-    Future.delayed(
-      const Duration(milliseconds: 250),
-      () => _scaffoldStateKey.currentState!.closeEndDrawer(),
-    );
-  }
+    // initialize parameters.
+    final reverseGeocodeParameters = ReverseGeocodeParameters()..maxResults = 1;
 
-  void onSearchSubmitted(String searchText) async {
-    final geocodeResults = await _locatorTask.geocode(
-      searchText: searchText,
-      parameters: _geocodeParameters,
+    // reverse geocode.
+    final reverseGeocodeResult = await _locatorTask.reverseGeocode(
+      location: normalizedTapPoint as ArcGISPoint,
+      parameters: reverseGeocodeParameters,
     );
+    if (reverseGeocodeResult.isEmpty) return;
 
-    if (geocodeResults.isEmpty) return;
+    // get and show the address.
+    final firstResult = reverseGeocodeResult.first;
+    final cityString = firstResult.attributes['City'] ?? "" as String;
+    final addressString = firstResult.attributes['Address'] ?? "" as String;
+    final stateString = firstResult.attributes['RegionAbbr'] ?? "" as String;
+    List<String> resultString = [addressString, cityString, stateString];
+    String combinedString =
```

Comment:
As a coding style, we want to use type inference whenever we can, so these can be `final resultString =` and `final combinedString =`

Reply:
and actually `resultString` should be `resultStrings` since it's a list

Reply:
Okay! Is there a specific reason why we use type inference. To reduce human error?

Reply:
Basically let the compiler do the work. Less typing. And like, if you later change the type of the thing on the right side of the equation, you don't also have to do the bookkeeping on the left side to keep up -- the type will propagate wherever it needs to go and the compiler will tell you if it runs into a mismatch somewhere.

Reply:
That makes sense, thanks Paul!

Reply:
ohh that makes sense, thanks for the explanation!